core/src/test/java/org/apache/iceberg/TestRowLineageMetadata.java

```diff
@@ -241,85 +257,110 @@ public void testDeletes() {
     table.newDelete().deleteFile(file).commit();
 
     // Deleting a file should create a new snapshot which should inherit last-row-id from the
-    // previous metadata and not
-    // change last-row-id for this metadata.
+    // previous metadata and not change last-row-id for this metadata.
     assertThat(table.currentSnapshot().firstRowId()).isEqualTo(30);
     assertThat(table.currentSnapshot().addedRows()).isEqualTo(0);
     assertThat(table.ops().current().nextRowId()).isEqualTo(30);
   }
 
   @TestTemplate
-  public void testReplace() {
-    assumeThat(formatVersion).isGreaterThanOrEqualTo(TableMetadata.MIN_FORMAT_VERSION_ROW_LINEAGE);
+  public void testPositionDeletes() {
```

Comment:
@RussellSpitzer, I added tests for position and equality deletes. These were previously failing because the record counts from delete manifests were also used. Please take a look at the new test cases and the fix.